lib/active_model/serializer/reflection.rb

```diff
@@ -87,8 +87,8 @@ def initialize(*)
       #       meta ids: ids
       #     end
       #   end
-      def link(name, value = nil)
-        options[:links][name] = block_given? ? Proc.new : value
+      def link(name, value = nil, &block)
+        options[:links][name] = block_given? ? block : value
```

Comment:
seems out of scope

Reply:
ruby 2.7 makes separation of positional and keyword arguments, which throws deprecation warning. it will eventually be dropped in ruby 3, in which case current code is likely to break.

so i think we can keep this change

Reply:
I'm at a loss to see the relationship between the argument changes in Ruby 2.7 and this diff. I believe that you saw a deprecation and that this made it go away.. but the reasoning sounds off

Let's get the 'out of scope' stuff extracted from this PR and merged before returning to this pr.

Reply:
ok. will do it soon